AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/Support/MutationEvent+Extensions.swift

```diff
@@ -10,16 +10,21 @@ import Dispatch
 import AWSPluginsCore
 
 extension MutationEvent {
-
-    // Updates the head of pending mutation event queue for a given model `id`
-    // if it has a `nil` version, with syncMetadata version in `mutationSync`
-    // and saves it in the mutation event table
-    static func updatePendingMutationEventVersionIfNil(for modelId: Model.Identifier,
-                                                       mutationSync: MutationSync<AnyModel>,
-                                                       storageAdapter: StorageEngineAdapter,
-                                                       completion: @escaping DataStoreCallback<Void>) {
+    // Reconciles the version of pending mutation events in mutation event table with the
+    // API response version for the following consecutive update scenarios :
```

Comment:
why do we reconcile? Consecutive operations that modify a model results in a sequence of pending mutation events that have the current version of the model. The first mutation event has the correct version of the model, while the subsequent events will have lower versions if the first mutation event is synced to the cloud. By checking reconciling the pending mutation events after syncing the first mutation event, we attempt to update the pending version to the latest version of the object.